src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityService.kt

```diff
@@ -106,12 +108,16 @@ class ActivityService(
       existingActivity.courseScheduleRules.addAll(newRules)
     }
 
+    activityRepository.saveAndFlush(existingActivity)
+
```

Comment:
I gave in and now save the entity before the @Transactional annotation because I need to know the ids of any saved schedule rules so I can use them in telemetry.